src/main/java/org/carlmontrobotics/robotcode2023/subsystems/Arm.java

```diff
@@ -15,148 +15,142 @@
-        GoalPos.LOW[1].wristPos = SmartDashboard.getNumber("LowWristCone", GoalPos.LOW[1].wristPos);
-        GoalPos.MID[1].armPos = SmartDashboard.getNumber("MidArmCone", GoalPos.MID[1].armPos);
-        GoalPos.MID[1].wristPos = SmartDashboard.getNumber("MidWristCone", GoalPos.MID[1].wristPos);
-        GoalPos.HIGH[1].armPos = SmartDashboard.getNumber("HighArmCone", GoalPos.HIGH[1].armPos);
-        GoalPos.HIGH[1].wristPos = SmartDashboard.getNumber("HighWristCone", GoalPos.HIGH[1].wristPos);
-        GoalPos.STORED[1].armPos = SmartDashboard.getNumber("StoredArmCone", GoalPos.STORED[1].armPos);
-        GoalPos.STORED[1].wristPos = SmartDashboard.getNumber("StoredWristCone", GoalPos.STORED[1].wristPos);
-        GoalPos.SHELF[1].armPos = SmartDashboard.getNumber("ShelfArmCone", GoalPos.SHELF[1].armPos);
-        GoalPos.SHELF[1].wristPos = SmartDashboard.getNumber("ShelfWristCone", GoalPos.SHELF[1].wristPos);
-        GoalPos.SUBSTATION[1].armPos = SmartDashboard.getNumber("SubstationArmCone", GoalPos.SUBSTATION[1].armPos);
-        GoalPos.SUBSTATION[1].wristPos = SmartDashboard.getNumber("SubstationWristCone", GoalPos.SUBSTATION[1].wristPos);
-        GoalPos.INTAKE[1].armPos = SmartDashboard.getNumber("IntakeArmCone", GoalPos.INTAKE[1].armPos);
-        GoalPos.INTAKE[1].wristPos = SmartDashboard.getNumber("IntakeWristCone", GoalPos.INTAKE[1].wristPos);
-
-        MAX_FF_VEL = SmartDashboard.getNumber("Max FF Vel", MAX_FF_VEL);
-        MAX_FF_ACCEL = SmartDashboard.getNumber("Max FF Accel", MAX_FF_ACCEL);
-    }
-
-    public void setArmVoltage(double volts) {
+    public void driveArm(double vel, double accel) {
+        double armFeedVolts = getKg() * getCoM().getAngle().getCos() + armFeed.calculate(vel, accel);
+        double armPIDVolts = armPID.calculate(armEncoder.getPosition(), goalPosRad[ARM]);
+        if ((getArmPos() > ARM_UPPER_LIMIT_RAD && armFeedVolts > 0) || 
+            (getArmPos() < ARM_LOWER_LIMIT_RAD && armFeedVolts < 0)) {
+            armFeedVolts = getKg() * getCoM().getAngle().getCos() + armFeed.calculate(0, 0);
+        }
+        // TODO: REMOVE WHEN DONE WITH TESTING (ANY CODE REVIEWERS, PLEASE REJECT MERGES
+        // TO MASTER IF THIS IS STILL HERE)
+        SmartDashboard.putNumber("ArmFeedVolts", armFeedVolts);
+        SmartDashboard.putNumber("ArmPIDVolts", armPIDVolts);
+        double volts = armFeedVolts + armPIDVolts;
+        SmartDashboard.putNumber("ArmTotalVolts", volts);
         armMotor.setVoltage(volts);
     }
-    public void setWristVoltage(double volts) {
+
+    public void driveWrist(double vel, double accel) {
+        double wristFeedVolts = wristFeed.calculate(getWristPosRelativeToGround(), vel, accel);
+        double wristPIDVolts = wristPID.calculate(getWristPos(), goalPosRad[WRIST]);
+        if ((getWristPos() > WRIST_UPPER_LIMIT_RAD && wristFeedVolts > 0) || 
+            (getWristPos() < WRIST_LOWER_LIMIT_RAD && wristFeedVolts < 0)) {
```

Comment:
This logic isn't correct. even if we're trying to move the arm the correct direction, wristFeedVolts may be > 0 due to kG

Reply:
replaced with vel <> 0